ports/unix/mpthreadport.c

```diff
@@ -85,8 +100,14 @@ void mp_thread_init(void) {
     thread->ready = 1;
     thread->arg = NULL;
     thread->next = NULL;
-    sem_init(&thread_signal_done, 0, 0);
 
+#if defined (__APPLE__)
+    GEN_SEM_NAME(thread_semaphore_name);
+    sem_t * thread_signal_done_p = &thread_signal_done;
+    thread_signal_done_p = sem_open(thread_semaphore_name, O_CREAT | O_EXCL, 0666, 1);
```

Comment:
I don't think this will work: the calls to `sem_wait()` and `sem_post()` will not use this pointer.  So there's going to need to be configuration for all bits of the code that use semaphores.  In the case of `__APPLE__` the global variable should be `STATIC sem_t *thread_signal_done_p;`, and in the rest of the cases just what it already is.

Reply:
@dmazzella it's close, but still needs a bit more work: `sem_open()` returns a pointer which then needs to be used elsewhere.  Eg:
```
sem_post(&thread_signal_done)  -->  sem_post(thread_signal_done_p)
sem_wait(&thread_signal_done)  -->  sem_wait(thread_signal_done_p)
```

Reply:
done in last commit